src/lib.rs

```diff
@@ -313,8 +308,8 @@ where
     RC: RollingCondition,
 {
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
-        let now = Local::now();
-        self.write_with_datetime(buf, &now)
+        let now = OffsetDateTime::now_local().unwrap_or_else(|_| OffsetDateTime::now_utc());
```

Comment:
FYI: This change can introduce a memory leak due to https://github.com/time-rs/time/issues/651

Reply:
A mach ports leak is better than a segfault, but this is something best addressed via a PR to the `num_threads` crate.

Reply:
This is called quite frequently and every call will leak at least `numberOfThreads` mach ports. This can easily accumulate to several GB of leaked memory for long-running processes like daemons.

It's also worth pointing out that `OffsetDateTime::now_local()` will always fail for macOS (unless the process is single threaded). Which makes this equivalent to `let now = OffsetDateTime::now_utc();` (on Apple platforms).
It might make sense to consider rotating based on UTC instead of local time in general. This would save quite a few CPU cycles and it's what would happen on Apple platforms with the suggested code anyway.

Reply:
I think rotating on UTC is a totally acceptable solution. I'm happy to modify the PR, but I haven't heard anything from the maintainer in over a year so I'm not sure if the crate is still being maintained.